projects/mahadao/bsc.js

```diff
@@ -1,115 +1,219 @@
+        sdk.api.abi.call({
+          target: trove,
+          abi: getEntireSystemCollAbi,
+          block,
+          chain,
+        })
+      )
+    );
+
+    COLLATERAL_ADDRESSES.forEach(async (collateral, index) => {
+      const collateralKeys = Object.keys(collateral);
+      let key = chain + ":" + collateral[collateralKeys[0]][0];
+      let val = tvls[index].output;
+
+      if (coingeckoIds[index]) key = coingeckoIds[index];
+
+      if (ret[key] == undefined) ret[key] = BigNumber(0);
+
+      if (decimals[index] !== undefined)
+        val = Number(val) / 10 ** decimals[index];
+
+      ret[key] = ret[key].plus(BigNumber(val));
+    });
+
+    return ret;
+  };
+}
+
+const getBalance = async (target, owner, block) => {
+  return (
+    await balanceOf({
+      target: target,
+      owner: owner,
+      block,
+      chain,
+    })
+  ).output;
+};
+
+const getTotalSupply = async (target, block) => {
+  return (
+    await totalSupply({
+      target: target,
+      block,
+      chain,
+    })
+  ).output;
+};
+
+async function getTVLOfarthu3ps(balances, block) {
+  const arthuBal = await getBalance(
+    bsc["arth.usd"],
+    bsc["ellipsis.arthu3eps.pool"],
+    block
+  );
+  const epsBal = await getBalance(
+    bsc["ellipsis.3eps.token"],
+    bsc["ellipsis.arthu3eps.pool"],
+    block
+  );
+
+  const token2Amount = new BigNumber(epsBal).dividedBy(e18);
+
+  sdk.util.sumSingleBalance(balances, "arth.usd", arthuBal);
+  sdk.util.sumSingleBalance(balances, "usd-coin", token2Amount.toNumber()); // todo: need to break this down
 }
 
 const replaceMAHAonBSCTransform = (addr) => {
   if (addr.toLowerCase() === "0xce86f7fcd3b40791f63b86c3ea3b8b355ce2685b")
     return "mahadao";
+
+  if (addr.toLowerCase() === "0xb69a424df8c737a122d0e60695382b3eec07ff4b")
+    return "arth";
   return `bsc:${addr}`;
 };
 
+const getTVLOfarthuval3ps = async (balances, block) => {
+  //get balance of arth-usd of arth-usd+val3eps
+  const arthUSDBalance = await getBalance(
+    bsc["arth.usd"],
+    bsc["ellipsis.arthuval3ps.pool"],
+    block
+  );
+
+  sdk.util.sumSingleBalance(balances, "arth.usd", arthUSDBalance);
+
+  // get balance of val3eps of arth-usd+val3eps
+  const val3EPSowned = await getBalance(
+    bsc["ellipsis.val3eps.token"],
+    bsc["ellipsis.arthuval3ps.pool"],
+    block
+  );
+
+  // todo hack for now
+  sdk.util.sumSingleBalance(balances, "usd-coin", val3EPSowned / e18);
+  return;
+};
+
 function pool2s() {
   return async (_timestamp, _ethBlock, chainBlocks) => {
     const balances = {};
 
-    // calculate tvl for regular uniswap lp tokens
-    const stakingContracts = [bsc.arthBusdStaking, bsc.arthMahaStaking];
-    const lpTokens = [bsc.arthBusdLP, bsc.arthMahaLP];
-    await sumTokensAndLPsSharedOwners(
+    await sumTokens(
       balances,
-      lpTokens.map((token) => [token, true]),
-      stakingContracts,
+      [
+        // apeswap ARTH/MAHA
+        [bsc.arth, bsc["apeswap.arthMahaLP"]],
```

Comment:
are you adding up all the tokens in the LP as pool2, if so, that is not the right approach. Can you provide the masterchef contract address, I can help with fetching correct pool2

Reply:
sorry, this needs a bit of rework, so gonna merge this PR and work on it.